### PR TITLE
T197148 Generate wikitext for on-wiki discussions

### PR DESCRIPTION
**Pull request by davidbarratt**
and @aezell !

This change is ontop of #95 so that PR will need to be merged before this one.

### CODE REVIEW

**Comment by davidbarratt** [comment]
This is ready for review now. :)

**Comment by davidbarratt** [comment]
This is ready to be tested!

File: i18n/en.json [hunk 1]
```diff
@@ -29,5 +29,10 @@
 		"report-bug": "Report a bug",
 		"revision-edit-summary-removed": "edit summary removed",
 		"view-source-code-on": "View on $1",
-		"warning-no-results": "No Results"
+		"warning-no-results": "No Results",
+		"discuss-on-wiki": "Share these results",
+		"discuss-on-wiki-text": "Between $1 and $2 on $3 [[User:$4]] ([[User talk:$4|talk]] | ([[Special:Contributions/$4|contribs]]) and [[User:$5]] ([[User talk:$5|talk]] | ([[Special:Contributions/$5|contribs]]) both made edits to some pages in common. You can see a chronological list of all their edits on the [$6 Interaction Timeline] or a table view on the [$7 Interaction Analyser].",
```
Comment (dayllanmaza) [inline i18n/en.json]: `Between 2000-01-01 and 2018-08-02 on en.wikipedia.org User:DMaza (talk | (contribs) and ...`
I think there is an unnecessary  `(` before the contribs link

File: i18n/en.json [hunk 2]
```diff
@@ -29,5 +29,10 @@
...
+		"discus-on-wiki-help": "Copy & paste this wikitext to an on-wiki discussion to share these Interaction Timeline results with others",
+		"discus-on-wiki-copy": "Copy",
```
Comment (dayllanmaza) [inline i18n/en.json]: typos on `discus`

File: i18n/qqq.json [hunk 3]
```diff
@@ -35,5 +35,10 @@
 	"report-bug": "Report a bug link.",
 	"revision-edit-summary-removed": "Message when the edit summary was removed",
 	"view-source-code-on": "Where to view the source code: View on GitHub",
-	"warning-no-results": "A warning that displays when there are no results"
+	"warning-no-results": "A warning that displays when there are no results",
+	"discuss-on-wiki": "Button that is displayed when results are present to prompt user to discuss the reuslts on wiki.",
+	"discuss-on-wiki-text": "The text that is made avilable to copy to a wiki. Variables $1 and $2 are dates, variable $3 is a wiki domain name, $4 and $5 are usernames, $6 and $7 are urls.",
+	"discus-on-wiki-help": "Instructions on what to do with the discuss-on-wiki-text that is presented.",
```
Comment (dayllanmaza) [inline i18n/qqq.json]: typos `discus`

File: i18n/qqq.json [hunk 4]
```diff
@@ -35,5 +35,10 @@
...
+	"discus-on-wiki-copy": "Call to action in discss on wiki modal to copy the wikitext to the clipboard.",
```
Comment (dayllanmaza) [inline i18n/qqq.json]: typo `discss`

File: client/styles/styles.scss [hunk 5]
```diff
@@ -331,3 +331,17 @@ footer {
 .sk-three-bounce {
 	margin: 0;
 }
+
+.btn-share {
+	position: fixed;
+	right: 2%;
+	bottom: 2%;
+	visibility: visible;
```
Comment (dayllanmaza) [inline client/styles/styles.scss]: You can probably remove this line; visible is the default value for visibility afaik

File: client/src/components/share/share.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+		const buttonClassName = [
+			'btn-share',
+			this.props.empty ? 'empty' : ''
+		];
+
+		const copiedMessage = this.state.copied ? 'discus-on-wiki-copied' : 'discus-on-wiki-copy';
+
+		return (
+			<React.Fragment>
+				<Button color="light" className={buttonClassName.join( ' ' )} onClick={this.toggle}>
```
Comment (dayllanmaza) [inline client/src/components/share/share.js]: The button overlaps with the footer logo "powered by wikimedia toolforge" when there are only a few interactions or if you scroll all the way down where the footer is visible
Eg: `/?wiki=enwiki&user=75.174.110.184&user=2601%3A1c1%3Ac100%3Aac50%3A781f%3A195d%3A7496%3A2849&startDate=1482710400`

**Review by dayllanmaza — CHANGES_REQUESTED** [review]
Other than these tiny things, it looks awesome.

**Comment by dayllanmaza** [comment]
BTW, I still think we should have as few commits as possible per pull request (ideally 1). Specially if you don't like squashing. 
This will make it easier to track down things (in the rare need of such event). Also easier to roll back since you can easily find the previous issue

**Comment by davidbarratt** [comment]
@dayllanmaza yeah this normally wouldn't have happened, but I had to merge the other issue into this one for testing.

**Comment by davidbarratt** [comment]
@dayllanmaza I've made all the requested changes, it should be good to review again.

**Review by dayllanmaza — APPROVED** [review]

**Pull request merged**